src/controllers/auth.controller.js

```diff
@@ -0,0 +1,325 @@
+import { dynamicSchema } from './../utils/dynamicSchema.js';
+import bcrypt from 'bcrypt';
+
+const register = async (req, res) => {
+  const userInformation = req.body;
+
+  const { error, value } = userSchema.validate(userInformation, {
+    allowUnknown: true,
+  });
+
+  const hashedPassword = await bcrypt.hash(value.password, 10);
+
+  if (error) {
+    throw ApiError.badRequest(error.message, { field: error.message });
+  }
+
+  await usersService.register({ ...value, password: hashedPassword });
+
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+
+  const user = await usersService.getOne({ activationToken });
+
+  if (!user) {
+    throw ApiError.notFound({ user: 'User not found' });
+  }
+
+  await usersService.update({ activationToken }, { activationToken: null });
+
+  res.send({ message: 'OK' });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await usersService.getOne({ email });
+
+  if (!user) {
+    throw ApiError.unauthorized({ user: 'invalid email or password' });
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.unauthorized({ user: 'invalid email or password' });
+  }
+
+  if (user.activationToken) {
+    throw ApiError.unauthorized({ user: 'profile is not activated' });
+  }
+
+  generateTokens(res, user);
+};
+
+const getUser = async (req, res) => {
+  const { userId } = req;
+
+  const user = await usersService.getOne({ id: userId });
+
+  res.json(usersService.normalize(user));
+};
+
+const resetPassword = async (req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw ApiError.badRequest('email is required', {
+      email: 'email is required',
+    });
+  }
+
+  const user = await usersService.getOne({ email });
+
+  if (!user) {
+    throw ApiError.badRequest('This email is not registered', {
+      email: 'This email is not registered',
+    });
+  }
+
+  const resetPasswordToken = jwtService.signConfirmToken(
+    usersService.normalize(user),
+  );
+
+  await usersService.update({ email }, { resetPasswordToken });
+
+  await emailService.sendResetPasswordEmail(email, resetPasswordToken);
+
+  res.send({ message: 'Check your email for reset password' });
+};
+
+const changePassword = async (req, res) => {
+  const { resetPasswordToken } = req.params;
+  const { password, confirmPassword } = req.body;
+
+  const user = jwtService.verify(resetPasswordToken);
+
+  if (!user) {
+    await usersService.update(
+      { resetPasswordToken },
+      {
+        resetPasswordToken: null,
+      },
+    );
+
+    throw ApiError.badRequest('The provided token is no longer valid', {
+      token: 'The token has expired or is invalid',
+    });
+  }
+
+  if (password !== confirmPassword) {
+    throw ApiError.badRequest('Passwords are differents', {
+      password: 'Passwords are differents',
+    });
```

Comment:
The error message 'Passwords are differents' contains a grammatical error. Consider changing it to 'Passwords are different'.